### PR TITLE
Quit always giving heart attacks to people on the Interlink like some kind of RP-seeking missile

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

Keep the heart attack event off the Interlink

## How This Contributes To The Skyrat Roleplay Experience

Staff sanity having to always aheal

## Proof of Testing

Can't really. We'll just have to fuck around and find out. (The double entendre is real.)

## Changelog

:cl: LT3
code: Engaging in Role Play on the Interlink should no longer lead to heart attack and death
/:cl:

### CODE REVIEW

**Comment by burgerenergy** [comment]
It's not technically in scope, but people get appendix issues in the Ghost Cafe from time to time, too, if you want to hit that in the same pass.

**Comment by lessthnthree** [comment]
> It's not technically in scope, but people get appendix issues in the Ghost Cafe from time to time, too, if you want to hit that in the same pass.

Sure

File: modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/proc/engaged_role_play_check(mob/living/carbon/human/player)
+	var/turf/player_turf = get_turf(player)
+	var/area/player_area = get_area(player_turf)
+	if(!is_station_level(player_turf.z))
+		return TRUE
+
+	if(istype(player_area, /area/station/commons/dorms))
```
Comment (Iamgoofball) [inline modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm]: Is dorms back in the protected spaces on-station? I thought we un-protected dorms in terms of "being on the station puts you at risk of un-targeted stuff"

Comment (lessthnthree) [reply modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm]: it's protected for like antags and disasters and stuff. but realistically if an erp'er gets appendicitis it gets ahealed. and disease outbreak is a waste of an event credit if it just infects someone locked in a dorm and also has to be manually rerolled

as discussed on discord we'll add args to determine what protections it checks

Comment (Iamgoofball) [reply modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm]: 👍

File: modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Checks if a player meets certain conditions to exclude them from event selection.
+ */
+/proc/engaged_role_play_check(mob/living/carbon/human/player, station = TRUE, dorms = TRUE)
+	var/turf/player_turf = get_turf(player)
+	var/area/player_area = get_area(player_turf)
+	if(!is_station_level(player_turf.z) && station)
+		return TRUE
```
Comment (Ghommie) [inline modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm:8]: This should check that the player is at the Interlink or similar places, and not anywhere off-station (including lavaland and deep space)

File: modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+	if(istype(player_area, /area/station/commons/dorms) && dorms)
```
Comment (Ghommie) [inline modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm:10]: In the face of the above suggestion, the deep space hotel should ward off those random events.

Comment (Iamgoofball) [reply modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm:10]: is station level check is already hitting that

Comment (Ghommie) [reply modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm:10]: Right now, yes, but should LT3 change the off-station check to an on-interlink / ghost cafe check that won't be longer the case, Goof.

Comment (lessthnthree) [reply modular_skyrat/modules/ices_events/code/events/ev_roleplay_check.dm:10]: We can add more granularity, but I don't want disease outbreak wasting itself anywhere off station.

**Comment by burgerenergy** [comment]
Another small thing; Interdyne on Icebox is technically within the station, as it's on the same z-level and spaghetti code. Events will strike them, as often they do (fucking solar flares.) Given that Dyne is usually two screens away from the NT station I don't see this as an issue and, at worst, it's a fun bit of interaction that comes up from time to time, but if you're trying to squash all the "off-station" events I'll bring your attention to it.

Full disclosure; I don't really want that changed, but!

**Comment by lessthnthree** [comment]
> Another small thing; Interdyne on Icebox is technically within the station, as it's on the same z-level and spaghetti code. Events will strike them, as often they do (fucking solar flares.) Given that Dyne is usually two screens away from the NT station I don't see this as an issue and, at worst, it's a fun bit of interaction that comes up from time to time, but if you're trying to squash all the "off-station" events I'll bring your attention to it.
> 
> Full disclosure; I don't really want that changed, but!

I don't recall disease hitting them because it's a ghost role and so it's checked. CMEs don't check for a particular person so they get hit. I don't particularly care to change this either. :v

Nothing stops Interdyne from neutralizing a CME.

**Review by vinylspiders — APPROVED** [review]

**Pull request merged**